src/qt/radiobut.cpp

```diff
@@ -39,11 +42,52 @@ bool wxRadioButton::Create( wxWindow *parent,
              const wxString& name)
 {
     m_qtRadioButton = new QRadioButton( parent->GetHandle() );
+
+    if ( (style & wxRB_GROUP) || (style & wxRB_SINGLE) )
+    {
+        if ( style & wxRB_SINGLE )
+        {
+            // Ensure that other buttons cannot join the last existing group
+            m_lastGroup.erase(parent);
+        }
+        else
+        {
+            QButtonGroup* qtButtonGroup = new QButtonGroup();
```

Comment:
I don't understand where is this pointer deleted. We don't seem to do it nor give its ownership to Qt. Are you sure there is no memory leak here?

Reply:
Yes you're correct, there is a memory leak here. One solution seems to be to delete the pointer in the descructor before erasing the element from the map.
 
As much as I'd like to use smart pointers, I know that we're aiming to keep compatibility with C++03 compilers. From looking at QT's offering, the only applicable one we could "wrap" this up in is `QSharedPointer` but I feel that's misleading as this is not meant to be shared (`QScoped' pointer is not compatible with available containers).
 
What do you think?

Reply:
What if you simply pass `parent->GetHandle()` as the parent of `qtButtonGroup`?